### DISCUSSION TITLE
4.5.6升级到4.5.11报错

### DISCUSSION THREAD

**Discussion by kggg** [rbatis/rbatis#483] [Q&A]
升级之前运行正常， 升级后报错， 麻烦看看哪里的原因？

`

steven@steven:~/data/rust/orm_test$ cargo run -p axum_test
   Compiling rbdc-pool-fast v4.5.6
error[E0407]: method `set_timeout` is not a member of trait `Pool`
  --> /home/steven/data/rust/.cargo/registry/src/rsproxy.cn-8f6827c7555bfaf8/rbdc-pool-fast-4.5.6/src/lib.rs:83:5
   |
83 |       async fn set_timeout(&self, timeout: Option<Duration>) {
   |       ^        ----------- help: there is an associated function with a similar name: `get_timeout`
   |  _____|
   | |
84 | |         self.timeout.store(timeout);
85 | |     }
   | |_____^ not a member of trait `Pool`

error[E0308]: mismatched types
   --> /home/steven/data/rust/.cargo/registry/src/rsproxy.cn-8f6827c7555bfaf8/rbdc-pool-fast-4.5.6/src/lib.rs:120:34
    |
120 |         let r = self.inner.check(conn).await;
    |                            ----- ^^^^ expected `ConnectionBox`, found `&mut ConnectionBox`
    |                            |
    |                            arguments to this method are incorrect
    |
note: method defined here
   --> /home/steven/data/rust/.cargo/registry/src/rsproxy.cn-8f6827c7555bfaf8/rbdc-4.5.11/src/pool/conn_manager.rs:58:18
    |
58  |     pub async fn check(&self, mut conn: ConnectionBox) -> Result<ConnectionBox, Error> {
    |                  ^^^^^

Some errors have detailed explanations: E0308, E0407.
For more information about an error, try `rustc --explain E0308`.
error: could not compile `rbdc-pool-fast` (lib) due to 2 previous errors

`

**Comment by zhuxiujia** [comment]
> registry

registry的问题，不要用代理

**Reply by kggg** [reply]
改了registry， 还是一样

**Reply by lopo1** [reply]
解决了吗

**Reply by zhuxiujia** [reply]
换官方源，删Cargo.lock然后 cargo update

**Reply by kggg** [reply]
这个问题已经解决了， 没有及时回复。
 按之前 的经经验只要删除cargo.lock就可以， 还是走代理。

**Reply by lopo1** [reply]
我删除了./cargo/registry  删除了cargo.lock 还是不行  ：
 cargo run
   Compiling salvo-oapi v0.66.2
   Compiling rbdc-pool-fast v4.5.8
   Compiling thread_local v1.1.8
   Compiling bitflags v2.5.0
   Compiling dyn-clone v1.0.17
   Compiling either v1.10.0
error[E0407]: method `set_timeout` is not a member of trait `Pool`
  --> C:\Users\Administrator\.cargo\registry\src\index.crates.io-6f17d22bba15001f\rbdc-pool-fast-4.5.8\src\lib.rs:81:5
   |
81 |       async fn set_timeout(&self, timeout: Option<Duration>) {
   |       ^        ----------- help: there is an associated function with a similar name: `get_timeout`
   |  _____|
   | |
82 | |         self.timeout.store(timeout);
83 | |     }
   | |_____^ not a member of trait `Pool`

error[E0407]: method `begin` is not a member of trait `Connection` 这就奇怪了

**Reply by zhuxiujia** [reply]
rustup update 升级编译器版本了吗

**Reply by lopo1** [reply]
rustc 1.77.0 (aedd173a2 2024-03-17)  已经是最新的了还是不行

**Reply by zhuxiujia** [reply]
abs_admin可以跑，地址是https://github.com/rbatis/abs_admin

**Reply by lopo1** [reply]
跑起你这个项目，再跑我的demo，莫名其妙的又不报错了。。。

**Reply by zhuxiujia** [reply]
那就是有缓存，没更新库呗
